### DISCUSSION TITLE
DateTime Column type issue in new Dotmim Version

### DISCUSSION THREAD

**Discussion by lv1986** [Mimetis/Dotmim.Sync#861] [Q&A]
Hi, 
I have found that all my `DateTime` Column Types from SQL Server are automatically converted to type  `text` in SQLite database during sync. Do you have any idea why this is happening? In the previous version it is correctly converted to `datetime`.

I have updated the SQL Script File: [https://github.com/Mimetis/LV1986](https://github.com/Mimetis/LV1986)

**SQL Server**
![1](https://user-images.githubusercontent.com/66375806/201018656-34c0d458-523a-4f40-96cb-50e02468150b.png)

**Sqlite Database created after first sync**
![2](https://user-images.githubusercontent.com/66375806/201018682-6f03d871-bbd1-4dc5-b7b2-a35b12e4936e.png)

**Comment by lv1986** [comment]
Hi @Mimetis ,
Just following up on this. Did you get time to check this issue?

**Comment by Mimetis** [comment]
DMS is using the `Microsoft.Data.Sqlite.SqliteType` to make type conversion.
SQLite internal types are basically:

``` cs
 public enum SqliteType
 {
     /// <summary>
     ///     A signed integer.
     /// </summary>
     Integer = SQLITE_INTEGER,

     /// <summary>
     ///     A floating point value
     /// </summary>
     Real = SQLITE_FLOAT,

     /// <summary>
     ///     A text string.
     /// </summary>
     Text = SQLITE_TEXT,

     /// <summary>
     ///     A blob of data.
     /// </summary>
     Blob = SQLITE_BLOB
 }

```
You can read more here: https://learn.microsoft.com/en-us/dotnet/standard/data/sqlite/types.

That's been said, you can still create the empty client database with a Date typed columns in SQLite (which by the way will be converted to text internally, see here : https://www.sqlite.org/datatype3.html), if you prefer
